pkg/chain/ethlike/mining_waiter.go

```diff
@@ -114,36 +156,198 @@ func (mw MiningWaiter) ForceMining(
+		// with the maximum.
+		if gasPrice.Cmp(maxGasPrice) > 0 {
+			gasPrice = maxGasPrice
 		}
 
-		// transaction not yet mined and we are still under the maximum allowed
+		// Transaction not yet mined and we are still under the maximum allowed
 		// gas price; resubmitting transaction with 20% higher gas price
-		// evaluated earlier
+		// evaluated earlier.
 		logger.Infof(
-			"resubmitting previous transaction [%v] with a higher gas price [%v]",
+			"resubmitting previous transaction [%v] "+
+				"with a higher gas price [%v]",
 			transaction.Hash.TerminalString(),
 			gasPrice,
 		)
-		transaction, err = resubmitFn(gasPrice)
+		transaction, err = resubmitFn(
+			&ResubmitParams{
+				GasPrice: gasPrice,
+			},
+		)
 		if err != nil {
-			logger.Warningf("could not resubmit TX with a higher gas price: [%v]", err)
+			logger.Warningf(
+				"could not resubmit TX with a higher gas price: [%v]",
+				err,
+			)
 			return
 		}
 	}
 }
+
+func (mw *MiningWaiter) forceMiningDynamicFeeTx(
+	originalTransaction *Transaction,
+	resubmitFn ResubmitTransactionFn,
+) {
+	logger.Infof(
+		"starting mining waiter for dynamic fee transaction: [%v]",
+		originalTransaction.Hash.TerminalString(),
+	)
+
+	// If the original transaction's gas fee cap was higher or equal the max
+	// allowed we do nothing; we need to wait for it to be mined.
+	if originalTransaction.GasFeeCap.Cmp(mw.maxGasFeeCap) >= 0 {
+		logger.Infof(
+			"original transaction gas fee cap is higher than the max allowed; " +
+				"skipping resubmissions",
+		)
+		return
+	}
+
+	transaction := originalTransaction
+	for {
+		receipt, err := mw.waitMined(mw.checkInterval, transaction)
+		if err != nil {
+			logger.Infof(
+				"transaction [%v] not yet mined: [%v]",
+				transaction.Hash.TerminalString(),
+				err,
+			)
+		}
+
+		// Transaction mined, we are good.
+		if receipt != nil {
+			logger.Infof(
+				"transaction [%v] mined with status [%v] at block [%v]",
+				transaction.Hash.TerminalString(),
+				receipt.Status,
+				receipt.BlockNumber,
+			)
+			return
+		}
+
+		// Transaction not yet mined, if the previous gas fee cap was the
+		// maximum one, we no longer resubmit.
+		oldGasFeeCap := transaction.GasFeeCap
+		if oldGasFeeCap.Cmp(mw.maxGasFeeCap) == 0 {
+			logger.Infof(
+				"reached the maximum allowed gas fee cap; " +
+					"stopping resubmissions",
+			)
+			return
+		}
+
+		// Increase the gas tip cap by 20%. A minimum increase by 10% comparing
+		// to the previous value is required for transaction replacement to be
+		// accepted by miners as mentioned in:
+		// https://github.com/ethereum/go-ethereum/pull/22898/files#r636583352.
+		// We increase it even more than the required level to greatly increase
+		// the transaction's chance for being picked up by miners.
+		oldGasTipCap := transaction.GasTipCap
+		newGasTipCap := new(big.Int).Add(
+			oldGasTipCap,
+			new(big.Int).Div(oldGasTipCap, big.NewInt(5)),
```

Comment:
Can we add a comment this line is our `// + 20%`?

Reply:
See https://github.com/keep-network/keep-common/pull/84/commits/25579c5c